crates/abq_queue/tests/integration.rs

```diff
@@ -736,6 +742,23 @@ async fn run_test(server: Server, steps: Steps<'_>) {
     queue.shutdown().await.unwrap();
 }
 
+async fn sleep_until_or_fail<T, F, Fut>(f: F, max: Duration) -> T
```

Comment:
might call this `poll until`. looks good. Do you have an idea of why the polling is necessary?

Reply:
[... this flaked again ❓ ](https://github.com/rwx-research/abq/actions/runs/5145565323/jobs/9263367521)